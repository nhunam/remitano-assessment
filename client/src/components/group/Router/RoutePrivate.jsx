import { Redirect, Route } from "react-router-dom";

import PropTypes from "prop-types";
import React from "react";

const RoutePrivate = ({
  component: Component,
  isAuthenticated,
  to,
  path,
  ...rest
}) => (
  <Route
    {...rest}
    render={props =>
      isAuthenticated ? (
        <Component {...props} />
      ) : (
        <Redirect
          key={path}
          to={{
            pathname: to
          }}
        />
      )
    }
  />
);

RoutePrivate.propTypes = {
  component: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  location: PropTypes.object,
  to: PropTypes.string
};

RoutePrivate.defaultProps = {
  to: "/"
};

export default RoutePrivate;
