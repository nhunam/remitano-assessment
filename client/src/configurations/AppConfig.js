const AppConfig = {
  routePrefix: "",
  youtubeAPIKey: "AIzaSyB7mOeBlVf2sixfEqOAopTNTIYEbcweGjM",
  youtubeAPIURL: "https://www.googleapis.com/youtube/v3/videos"
};

export default AppConfig;
// export const baseUrl = process.env.NODE_ENV === 'development' ? window.location.href : 'https://obscure-beach-46869.herokuapp.com/'