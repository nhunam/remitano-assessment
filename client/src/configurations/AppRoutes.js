import { lazy } from "react";
/* istanbul ignore next */
const AppRoutes = [
  {
    path: "/video-sharing",
    exact: true,
    component: lazy(
      /* istanbul ignore next */ () => import("containers/screens/VideoSharing")
    )
  }
];

export default AppRoutes;
