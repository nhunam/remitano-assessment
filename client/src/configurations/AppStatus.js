const AppStatus = {
  IDLE: "idle",
  RUNNING: "running",
  READY: "ready",
  SUCCESS: "success",
  ERROR: "error"
};

export default AppStatus;
