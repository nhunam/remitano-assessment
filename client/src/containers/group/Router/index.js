import Router from "components/group/Router";

export default Router;
