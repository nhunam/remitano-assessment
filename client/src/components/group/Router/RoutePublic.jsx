import React from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';
import { AppConfig } from 'configurations';

export const RoutePublic = ({ component: Component, isAuthenticated, to, ...rest }) => (
  <Route
    {...rest}
    render={props => (isAuthenticated ? <Redirect to={to} /> : <Component {...props} />)}
  />
);

RoutePublic.propTypes = {
  component: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  to: PropTypes.string,
};

RoutePublic.defaultProps = {
  to: AppConfig.routePrefix,
};

export default RoutePublic;
