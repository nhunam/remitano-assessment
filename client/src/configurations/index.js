import AppRoutes from "./AppRoutes";
import AppConfig from "./AppConfig";
import AppStatus from "./AppConfig";

export { AppRoutes, AppConfig, AppStatus };
