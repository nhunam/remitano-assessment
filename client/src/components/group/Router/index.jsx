import React, { Suspense } from "react";

import { AppConfig } from "configurations";
import PropTypes from "prop-types";
import { Route } from "react-router-dom";
import RoutePrivate from "./RoutePrivate";
import RoutePublic from "./RoutePublic";
import { Switch } from "react-router";

const LoadingMessage = () => "I'm loading...";

function Router(props) {
  const { isAuthenticated, notFoundComponent, homeComponent, routes } = props;
  return (
    <Suspense fallback={<LoadingMessage />}>
      <Switch>
        {routes.map(route => {
          const { path, exact, component, ...rest } = route;
          return (
            <RoutePrivate
              key={`${AppConfig.routePrefix}${path}`}
              path={`${AppConfig.routePrefix}${path}`}
              to={`${AppConfig.routePrefix}/404`}
              isAuthenticated={isAuthenticated}
              component={component}
              {...rest}
            />
          );
        })}
        <Route path="/" exact component={homeComponent}></Route>
        <RoutePublic
          isAuthenticated={isAuthenticated}
          component={notFoundComponent}
        />
      </Switch>
    </Suspense>
  );
}

Router.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  routes: PropTypes.array.isRequired,
  notFoundComponent: PropTypes.any.isRequired,
  homeComponent: PropTypes.any.isRequired
};

export default Router;
