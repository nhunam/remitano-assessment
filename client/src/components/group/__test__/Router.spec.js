import HomeScreen from "containers/screens/Home";
import { MemoryRouter } from "react-router-dom";
import NotFoundScreen from "containers/screens/NotFound";
import { Provider } from "react-redux";
import React from "react";
import Router from "components/group/Router";
import configureMockStore from "redux-mock-store";
import { mount } from "enzyme";
import { AppRoutes as routes } from "configurations";

describe("Router", () => {
  const mockStore = configureMockStore();
  const store = mockStore({
    videos: {
      data: [],
      page: {
        total_pages: 1,
        has_next: false,
        has_previous: false,
        current_page: 0,
        total_elements: 0
      }
    }
  });
  it("should redirect for unauthenticated access", () => {
    const render = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/private"]}>
          <Router
            routes={routes}
            notFoundComponent={NotFoundScreen}
            homeComponent={HomeScreen}
            isAuthenticated={false}
          />
        </MemoryRouter>
      </Provider>
    );

    expect(render).toMatchSnapshot();
  });

  it("should redirect for authenticated access", () => {
    const render = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={["/private"]}>
          <Router
            routes={routes}
            notFoundComponent={NotFoundScreen}
            homeComponent={HomeScreen}
            isAuthenticated={true}
          />
        </MemoryRouter>
      </Provider>
    );

    expect(render).toMatchSnapshot();
  });
});
