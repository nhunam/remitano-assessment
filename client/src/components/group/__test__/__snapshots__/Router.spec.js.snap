// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router should redirect for authenticated access 1`] = `ReactWrapper {}`;

exports[`Router should redirect for unauthenticated access 1`] = `ReactWrapper {}`;
